### PR TITLE
Fix various SBOM creation problems

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
- Fix benign build ERROR with createSBOM jar SHA list
```
11:37:51  ERROR: Cannot determine jar version from /home/jenkins/workspace/build-scripts/jobs/jdk21u/jdk21u-linux-x64-temurin/sbin/../cyclonedx-lib/dependency_data/dependency_data.properties for SBOM creation dependency temurin-gen-cdxa.jar.
```
- Bom.metadata not being created correctly for addComponentProp, luckily it is the first method called, so didn't cause a problem!
- Rededundent addExternalReference and addComponentExtRef were not removed and unittests were actually failing, but they were no being tested with fail on error! so passed!
- Metadata.tools has been deprecated in CycloneDX 1.0.6 spec and will be removed at some point, replaced by ToolInformation.Component
- Add the 3 new extra CycloneDX 1.0.6 dependency jars to the SBOM classpath

### CODE REVIEW

**Comment by karianna** [comment]
@andrew-m-leonard linter fixes required

File: sbin/build.sh [hunk 1]
```diff
@@ -978,7 +979,7 @@ generateSBoM() {
   local fullVer=$(cat "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[TARGET_DIR]}/metadata/productVersion.txt")
   local fullVerOutput=$(cat "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[TARGET_DIR]}/metadata/productVersionOutput.txt")
 
-  # Create initial SBOM json
+  # Create initial SBOM json
```
Comment (sxa) [inline sbin/build.sh:982]: Extra space has appeared here
```suggestion
  # Create initial SBOM json
```

**Review by sxa — APPROVED** [review]
I trust you to have tested this so I'm not analyzing all the logic paths too closely in this review, but I suggest you accept the suggestion to remove the extra space before merging.

**Review by Haroon-Khel — APPROVED** [review]

**Comment by Haroon-Khel** [comment]
Ill rebase https://github.com/adoptium/temurin-build/pull/4075 with these changes once this pr gets merged

**Pull request merged**